presto-main/src/main/java/com/facebook/presto/operator/BigintGroupByHash.java

```diff
@@ -267,6 +273,14 @@ private void rehash()
         }
         int newCapacity = (int) newCapacityLong;
 
+        // An estimation of how much extra memory needed before allocating the hash table
+        // TODO let's think about how much memory we want to reserve 1X or 2X?
+        if (!updateReservedMemory.apply(((long) (newCapacity - hashCapacity)) * (Long.BYTES + Integer.BYTES))) {
```

Comment:
I generally prefer writing `x * (long) y` instead of `((long) x) * y`.

* Given all the brackets, whether it's correctly written is sometimes not obvious in the 2nd case.
* Sometimes this also saves a pair of bracket.